File: java64/nivel4/exer44.java
import javax.swing.JOptionPane;

public class exer44 {
    public static void main(String[] args) {
        int opcao;
        double celsius, fahrenheit, conversao, altura, pesoIdeal;
        String simNao;
        boolean continua = false;
        do {
            opcao = Integer.parseInt(JOptionPane.showInputDialog(
                    "Informe a opção\n1 – Conversão de Graus Celsius em Graus Fahrenheit\n2 – Conversão de Graus Fahrenheit em Graus Celsius\n3 – Peso ideal do homem\n4 – Peso ideal da mulher "));

            switch (opcao) {
            case 1:
                celsius = Double.parseDouble(JOptionPane.showInputDialog("Informe a temperatura em Celsius :"));
                conversao = (celsius * 9 / 5) + 32;
                JOptionPane.showMessageDialog(null, "A conversão de Graus Celsius " + celsius + "° em Graus Fahrenheit é: " + conversao);
                break;

            case 2:
                fahrenheit = Double.parseDouble(JOptionPane.showInputDialog("Informe a temperatura em Fahrenheit :"));
                conversao = (fahrenheit - 32) * 5 / 9;
                JOptionPane.showMessageDialog(null, "A conversão de Graus Fahrenheit  " + fahrenheit
                        + "° em Graus Celsius é: " + conversao);
                break;

            case 3:
                altura = Double.parseDouble(JOptionPane.showInputDialog("Informe a sua altura: "));
                pesoIdeal = (72.7 * altura) - 58;
                JOptionPane.showMessageDialog(null, "Peso ideal é: " + pesoIdeal);
                break;

            case 4:
                altura = Double.parseDouble(JOptionPane.showInputDialog("Informe a sua altura: "));
                pesoIdeal = (62.1 * altura) - 44.7;
                JOptionPane.showMessageDialog(null, "Peso ideal é: " + pesoIdeal);
                break;

            default:
                break;
            }
            simNao = JOptionPane.showInputDialog("Deseja encerrar o programa?");
            if (simNao == "S" || simNao == "s")
                continua = true;
        } while (continua = false);
    }
}

File: pyhton64/nivel4/exer44.py
def main():
    cont = True
    while cont:
        op = opcao()
        if op == 1:
            c_f()
        elif op == 2:
            f_c()
        elif op == 3:
            peso_homem()
        elif op == 4:
            peso_mulher()
        else:
            print('Opção invalida')

        op2 = input('S - s / N - n\nContinuar: ')
        if op2 == 'N' or op2 == 'n':
            cont = False


def opcao():
    opc = int(input('1 – Conversão de Graus Celsius em Graus Fahrenheit\n' +
                    '2 – Conversão de Graus Fahrenheit em Graus Celsius\n' +
                    '3 – Peso ideal do homem\n' +
                    '4 – Peso ideal da mulher\n' + 'Ecolha: '))
    return opc


def c_f():
    c = int(input('Informe o valor em graus Celsius para conversao: '))
    f = float(32 + ((9 * c) / 5))
    return print(f'Celsius para Fahrenheit: F = {f}')


def f_c():
    f = int(input('Informe o valor em graus Fahrenheit para conversao: '))
    c = float((f - 32) * 5 / 9)
    return print(f'Fahrenheit para Celsius: C = {c}')


def peso_homem():
    h = float(input('Informe sua altura: '))
    peso = (72.7 * h) - 58
    return print(f'Peso ideal é: {peso}')


def peso_mulher():
    h = float(input('Informe sua altura: '))
    peso = (62.1 * h) - 44.7
    return print(f'Peso ideal é: {peso}')


if __name__ == '__main__':
    main()
